pytest-embedded/pytest_embedded/plugin.py

```diff
@@ -1030,7 +1030,15 @@ def _fixture_classes_and_options(
 
                 classes[fixture] = App
         elif fixture == 'serial':
-            if 'esp' in _services:
+            if 'idf' in _services and target == 'linux':
```

Comment:
Here is not good enough. Here you're assuming the user would pass the target via CLI or param. actually it could also be read from the `IdfApp` instance.

Reply:
I think we need to use lazy load here.

in 

```python
@pytest.fixture
@multi_dut_generator_fixture
def serial(_fixture_classes_and_options, app) -> Optional['Serial']:
    """A serial subprocess that could read/redirect/write"""
    if 'serial' not in _fixture_classes_and_options.classes:
        return None

    cls = _fixture_classes_and_options.classes['serial']
    kwargs = _fixture_classes_and_options.kwargs['serial']
    if 'app' in kwargs and kwargs['app'] is None:
        kwargs['app'] = app
    return cls(**_drop_none_kwargs(kwargs))
```

here, we could get the `app` fixture, which is loaded already.

if the `app.target` == 'linux', we would use the `LinuxSerial` and `LinuxDut` as well.

Reply:
same for `def dut(...)`